tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionParserSuite.scala

```diff
@@ -21,6 +21,24 @@ import org.scalatest.FunSuite
 
 class RegularExpressionParserSuite extends FunSuite {
 
+  test("detect regexp strings") {
+    // Based on https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html
+    val strings: Seq[String] = Seq("\\", "\u0000", "\\x00",
+      "\f", "\\a", "\\e", "\\cx", "[abc]", "^", "[a-z&&[def]]", ".", "*", "\\d", "\\D",
+      "\\h", "\\H", "\\s", "\\S", "\\v", "\\V", "\\w", "\\w", "\\p", "$", "\\b", "\\B",
+      "\\A", "\\G", "\\Z", "\\z", "\\R", "?", "|", "(abc)", "a{1,}", "\\k", "\\Q", "\\E")
+    for (string <- strings) {
+      assert(RegexParser.isRegExpString(string))
+    }
+  }
+
+  test("detect non-regexp strings") {
+    val strings = Seq("\\.", "A", ",", "\t", ":", "")
```

Comment:
This looks wrong.  For example if I did a split with `"\\."` as the separator on `"a.b.c"` I would expect an output of `["a", "b", "c"]`. But I cannot take `"\\."` and pass it into a non-regexp string split, because it will try to look for a `"\\"` character in front of the '`.'`.  I am I wrong/missing something?  I think we could still support it, but we would have to transpile the regular expression into a constant string.

Reply:
Yes, I agree, this isn't what we want here. Moving this to draft for now and will rethink this approach.

Reply:
I updated to treat any string containing escaped characters as a regular expression.